src/main/java/com/neonlab/product/service/ProductService.java

```diff
@@ -93,11 +110,15 @@ private List<VarietyDto> saveAndMapVarieties(Product product, List<VarietyDto> v
     private void saveAndMapDocument(VarietyDto varietyDto, Variety variety) throws ServerException {
         if (!CollectionUtils.isEmpty(varietyDto.getDocuments())){
             var documents = documentService.saveAll(varietyDto.getDocuments());
-            for (var document : documents){
+            for (var document : documents) {
                 document.setDocIdentifier(variety.getId());
                 document.setEntityName(variety.getClass().getSimpleName());
                 documentService.save(document);
             }
+            var documentList = documentService.fetchByDocIdentifierAndEntityNameAsc(variety.getId(), variety.getClass().getSimpleName());
+            if (documentList.size() > 4) {
+                documentService.maintainSize(documentList);
+            }
```

Comment:
Common functionality can be used multiple places. Please make a method in document service (generic) for the same